### PR TITLE
I18n

### PR DESCRIPTION
**Pull request by andrestobelem**
# Summary
* I added i18n support for Spanish and English.
* Added a dropdown for language selecting.
* The routes of the blogs shows en language: http://localhost:3000/es/articles
* Lost in translation: Article.human_attribute_name(:title) is not working.

# Trello Card
https://trello.com/c/ctfuxybH

### CODE REVIEW

File: app/views/articles/index.html.slim [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
-= link_to 'New article', new_article_path
+= link_to t('articles.new.title'), new_article_path
 
-table
+table.table
   tr
-    th Title
-    th Text
+    th = Article.human_attribute_name('title')
```
Comment (epintos) [inline app/views/articles/index.html.slim]: This is not working?

I think i've never used this before

File: app/views/articles/show.html.slim [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-  strong Title:
+  strong = Article.human_attribute_name(:title)
   = @article.title
 p
-  strong Text:
+  strong = Article.human_attribute_name(:text)
   = @article.text
 
-h2 Comments        
+h2 = Article.model_name.human
```
Comment (epintos) [inline app/views/articles/show.html.slim]: This one works? I think I've used it before

File: config/routes.rb [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
   devise_for :users
-  get 'welcome/index'
-
-
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
 
-  resources :articles do
-    resources :comments
+  scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
```
Comment (epintos) [inline config/routes.rb]: If you are using the query param `locale` for the I18n, then this shouldn't be necessary

File: config/routes.rb [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
+      resources :comments
+    end
+
+    root 'welcome#index'
   end
+
   # You can have the root of your site routed with "root"
-  root 'welcome#index'
+
+  root  to: redirect("/#{I18n.default_locale}", status: 302),
```
Comment (epintos) [inline config/routes.rb]: Why this is necessary? As  I can see in your application controller you set the default locale if none is present

File: app/views/articles/index.html.slim (hunk 1 above)
Comment (andrestobelem) [inline app/views/articles/index.html.slim]: Nope, not working right now. I changed for another thing.

File: app/views/articles/show.html.slim (hunk 2 above)
Comment (andrestobelem) [inline app/views/articles/show.html.slim]: Yes, this one works.

File: config/routes.rb (hunk 3 above)
Comment (andrestobelem) [inline config/routes.rb]: Is just for adding /en/ or /es/ to the routes

File: config/routes.rb (hunk 4 above)
Comment (andrestobelem) [inline config/routes.rb]: Yes, I think that is not necessary.

File: config/routes.rb (hunk 3 above)
Comment (epintos) [inline config/routes.rb]: We don't use that in the routes to have nicer routes. Having the locale in the query param I think is enough

File: config/routes.rb [hunk 5]
```diff
@@ -13,8 +13,8 @@
 
   # You can have the root of your site routed with "root"
 
-  root  to: redirect("/#{I18n.default_locale}", status: 302),
-        as: :redirected_root
+  #root  to: redirect("/#{I18n.default_locale}", status: 302),
```
Comment (epintos) [inline config/routes.rb]: Its better to remove unused code

**Comment by epintos** [comment]
:+1:

**Pull request merged**